cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -294,48 +294,61 @@ sanityCheckElaboratedPackage ElaboratedConfiguredPackage{..}
 rebuildProjectConfig :: Verbosity
                      -> DistDirLayout
                      -> ProjectConfig
-                     -> IO (ProjectConfig,
-                            [PackageSpecifier UnresolvedSourcePackage])
-rebuildProjectConfig verbosity
-                     distDirLayout@DistDirLayout {
-                       distProjectRootDirectory,
-                       distDirectory,
-                       distProjectCacheFile,
-                       distProjectCacheDirectory
-                     }
-                     cliConfig = do
+                     -> IO
+                          ( ProjectConfig
+                          , [PackageSpecifier UnresolvedSourcePackage]
+                          )
+rebuildProjectConfig verbosity distDirLayout cliConfig = do
 
     (projectConfig, localPackages) <-
-      runRebuild distProjectRootDirectory $
-      rerunIfChanged verbosity fileMonitorProjectConfig () $ do
-
-        projectConfig <- phaseReadProjectConfig
-        localPackages <- phaseReadLocalPackages projectConfig
-        return (projectConfig, localPackages)
+      runRebuild distProjectRootDirectory
+      $ rerunIfChanged verbosity
+                       fileMonitorProjectConfig
+                       (projectConfigConfigFile, distProjectFile "")
```

Comment:
Don't pass the tuple directly. Instead make a new `fileMonitorProjectConfigValue` or `fileMonitorProjectConfigInput` or something.

Reply:
You mean a new datatype, right?

Reply:
I think tuple is ok here. 

I meant something like 
  ```
  let 
    fileMonitorProjectConfigInput = 
      (..., ...)
```